core/src/epicli/data/common/ansible/playbooks/roles/repository/tasks/setup.yml

```diff
@@ -74,6 +71,15 @@
     - not custom_repository_url
     - inventory_hostname in target_repository_hostnames
 
+# Put it down here as the script download-requirements.sh relies on some files
```

Comment:
I don't think this change makes any difference, that's because  the `enable-system-repos.sh` script depends on the `/var/tmp/enabled-system-repos.tar` file and the `/var/tmp/enabled-system-repos.tar`  file is created at the `- name: Set up repositories`. This means the `[[ -f /var/tmp/enabled-system-repos.tar ]] && [[ -f ${repos_backup_file} ]]` condition will evaluate to false anyway regardless of the changed order. 🤔

Reply:
In the `normal` situation your assumption is correct, that is the first time runs, no errors due to this.

However if the build fail, we re-run on existing cluster then I hit many times that the cyclic dependencies which I describe here. ssh in the system and found what I described in the change request.

No more investigation why and how the system ends up like this as no time. However consider the change wont do harm and most important, it fixes the re-run situation for me.

I am using this change and my system build reliability is sort of perfect now.

Reply:
Ok then. I did some manual testing and I don't see any harm caused by this. I'm OK with merging it. :+1:😇